docs/book/src/developer/providers/v1.3-to-v1.4.md

```diff
@@ -56,6 +56,7 @@ maintainers of providers and consumers of our Go API.
 
 ### Other
 
+- clusterctl describe cluster command now has a --show-machinehealthchecks flag.
```

Comment:
It's definitely fine to add this here, given what other things we already added before.

We should probably re-think which sort of things are actually relevant for provider authors